Assets/UI/WorldView/citybannermanager_CQUI_expansions.lua

```diff
@@ -0,0 +1,188 @@
+
+        if (localPlayerID == iCityOwner and self.CQUI_DistrictBuiltIM ~= nil) then
+            -- On first call into UpdateStats, CQUI_DistrictBuiltIM may not be instantiated yet
+            -- However this is called often enough that it's not a problem
+            -- Update the built districts 
+            self.CQUI_DistrictBuiltIM:ResetInstances(); -- CQUI : Reset CQUI_DistrictBuiltIM
+            local pCityDistricts:table = pCity:GetDistricts();
+            if (IsCQUI_SmartBanner_DistrictsEnabled()) then
+                for i, district in pCityDistricts:Members() do
+                    local districtType = district:GetType();
+                    local districtInfo:table = GameInfo.Districts[districtType];
+                    local isBuilt = pCityDistricts:HasDistrict(districtInfo.Index, true);
+                    if (isBuilt == true
+                        and districtInfo.Index ~= 0
+                        and districtInfo.DistrictType ~= "DISTRICT_GOVERNMENT"
+                        and districtInfo.DistrictType ~= "DISTRICT_WONDER"
+                        and districtInfo.DistrictType ~= "DISTRICT_CITY_CENTER") then
+                        SetDetailIcon(self.CQUI_DistrictBuiltIM:GetInstance(), "ICON_"..districtInfo.DistrictType);
+                    end
+                end
+            end
+        end
+    end
+end
+
+-- ============================================================================
+-- CQUI Replacement Functions
+-- ============================================================================
+function OnCityStrikeButtonClick( playerID, cityID )
+    print_debug("CityBannerManager_CQUI_Expansions: OnCityStrikeButtonClick ENTRY playerID:"..tostring(playerID).." cityID:"..tostring(cityID));
+    CQUI_OnCityRangeStrikeButtonClick(playerID, cityID);
+end
+
+-- ===========================================================================
+function OnDistrictRangeStrikeButtonClick( playerID, districtID )
+    print_debug("CityBannerManager_CQUI_Expansions: OnDistrictRangeStrikeButtonClick ENTRY playerID:"..tostring(playerID).." districtID:"..tostring(districtID));
+    local pPlayer = Players[playerID];
+    if (pPlayer == nil) then
+        return;
+    end
+
+    local pDistrict = pPlayer:GetDistricts():FindID(districtID);
+    if (pDistrict == nil) then
+        return;
+    end;
+
+    UI.DeselectAll();
+    UI.SelectDistrict(pDistrict);
+    -- CQUI (Azurency) : Look at the district plot
+    UI.LookAtPlot(pDistrict:GetX(), pDistrict:GetY());
+    UI.SetInterfaceMode(InterfaceModeTypes.DISTRICT_RANGE_ATTACK);
+end
+
+-- ===========================================================================
+--  CQUI Initialize Function
+-- ===========================================================================
+function Initialize()
+    print_debug("CityBannerManager_CQUI_Expansions: Initialize CQUI CityBannerManager")
+    -- Events are initialized in the common file
+    LateInitialize();
```

Comment:
You should not call LateInitialize() from Initialize().

Reply:
I need to rename lateinitialize, I think.  If I remember correct I modified citybannermanager does not declare its own LateInitialize to override, and it’s unclear to me how often one can call Initialize... this is an attempt to not have 3 different Initialize calls for the various CityBannerManager files
(unmodifed, citibannermanager_cqui, citybannermanager_cqui_(basegame/expansions))

Reply:
It does, it uses a different name. It is always hooked to ContextPtr:SetInitHandler(), and here it is OnContextInitialize. But there is no need to override that.